src/shims/foreign_items.rs

```diff
@@ -463,6 +463,11 @@ trait EvalContextExtPriv<'tcx>: crate::MiriInterpCxExt<'tcx> {
                     let size = this.read_target_usize(size)?;
                     let align = this.read_target_usize(align)?;
 
+                    // In rustc, the maximal supported alignment is 2^29.
+                    if align > 2u64.checked_pow(29).unwrap() {
+                        throw_unsup_format!("Alignment exceeded 2^29 is unsupported");
+                    }
+
                     this.check_rustc_alloc_request(size, align)?;
```

Comment:
Please add the check in check_rustc_alloc_request instead so that it is run on all entry points (you forgot reallocate).

Also, please add a test.

Reply:
Also is there a way to not hard-code the max value but use the same constant as rustc?

Reply:
> Also is there a way to not hard-code the max value but use the same constant as rustc?

Yup, we can check from the error type returned by ``Align::from_bytes`` https://github.com/rust-lang/rust/blob/8d0066922b14cee3175e8c141e5e909fa6d9a6eb/compiler/rustc_abi/src/lib.rs#L731